kong/core/handler.lua

```diff
@@ -519,11 +519,18 @@ return {
   },
   header_filter = {
     before = function(ctx)
+      local var = ngx.var
+      local header = ngx.header
+
       if ctx.KONG_PROXIED then
         local now = get_now()
         ctx.KONG_WAITING_TIME = now - ctx.KONG_ACCESS_ENDED_AT -- time spent waiting for a response from upstream
         ctx.KONG_HEADER_FILTER_STARTED_AT = now
       end
+
+      if singletons.configuration.server_tokens then
```

Comment:
This is tricky. Upon re-reading the description for both `server_tokens` and `latency_tokens`, this new header does not fall under either category. After giving it some thoughts, I have been thinking of changing how we specify those headers... bear with me.

As we strive towards a simpler configuration file (with less directive, but no less flexibility, and more intuitiveness), we can follow a similar approach as #3147, and provide a new way to configure such headers served by Kong, like so:

```
tokens = [server|latency|debug|off] custom ...
```

A few examples of which headers are served with each setting:
```
tokens = server
  Via: ... (or Server: ... upon non-proxying) - same behavior as server_tokens = on

tokens = latency
  X-Kong-Proxy-Latency: ...
  X-Kong-Upstream-Latency: ... (same behavior as latency_tokens = on

token = server, latency
  same as both previous examples

tokens = debug
  All previous headers plus:
  X-Kong-Upstream-Status
  X-Kong-Matched-API (already present in the router when the Kong-Debug request header is set)

tokens = X-Kong-Upstream-Latency
  Only send the upstream latency header

tokens = off
  No header will be injected by Kong core
```

I know this is more work than this PR originally intended to be, but I hope you can see where I am coming from here. We want to provide a configuration file that makes sense and is intuitive.